### ISSUE TITLE
An addon that makes the rotating input snap to 15-degree increments

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#5814]
### Why this would be helpful

When highlighting the input in the `point in direction` block, you get a popup with an arrow you can rotate to quickly set the direction without typing it in, and it snaps to 15° increments. There is another one of these in the direction input in the sprite properties area, but it doesn’t snap, unlike the one I previously mentioned.

### How this addon works

It may be useful to have that arrow snap to 15° increments too.

We could also let the user customize it, allowing it to be set to factors of 360, such as 1, 5, 15, and 45.

### Additional context

This repurposes #3699.

**Comment by BroJac5246** [comment]
If you're willing to go that far, then you're also probably willing to just type in the number of degrees.

**Comment by WorldLanguages** [comment]
@sfederici What do you think about this specific suggestion?

I'm a bit lost because discussion about this has been distributed between 4 different discussion/issues and it's hard to keep of everything said in all of them (#5809, #3726, #3699, and this issue)

**Comment by sfederici** [comment]
To me, a 15-degree increment is fine. As for the labels instead (if they are implemented), I guess limiting them to 0, 45, 90, 135, 180, -135, -90, -45 would be fine.

**Comment by WorldLanguages** [comment]
Anyone interested on summing up all suggestions and past PRs about rotation dials? I linked two issues and a failed PR in my comment above.

As far as I can remember, Scratch has two different rotation dial UI, one for the code area (AKA workspace) and another outside it. One of these two did snap to increments I think. Ideally, we should try to make them consistent, right?

**Comment by CST1229** [comment]
> As far as I can remember, Scratch has two different rotation dial UI, one for the code area (AKA workspace) and another outside it. One of these two did snap to increments I think. Ideally, we should try to make them consistent, right?

The code area rotation dial snaps to increments.

**Comment by BroJac5246** [comment]
> Anyone interested on summing up all suggestions and past PRs about rotation dials? I linked two issues and a failed PR in my comment above.

The suggestions are to provide a way to snap rotation inputs to either 45-degree or 15-degree angles, with this issue also suggesting 5-degree angles.

It seems that the consensus is that modifier keys should trigger or disable this behavior (for example, no key pressed could result in 1-degree increments, Shift could result in 15-degree increments, and Ctrl could be for 45-degree increments). Of course, this behavior would be customizable.

(lol that sounds AI-generated 😂)

> As far as I can remember, Scratch has two different rotation dial UI, one for the code area (AKA workspace) and another outside it. One of these two did snap to increments I think. Ideally, we should try to make them consistent, right?

That would be ideal and would be a positive side effect of this addon.

**Comment by WorldLanguages** [comment]
> It seems that the consensus is that modifier keys should trigger or disable this behavior (for example, no key pressed could result in 1-degree increments, Shift could result in 15-degree increments, and Ctrl could be for 45-degree increments). Of course, this behavior would be customizable.

Yeah, but one of the two rotation dial UIs in Scratch already snap every a certain amount of degrees... Would these settings apply to both?
I mean, it's just weird that disabling the "snap" setting would do nothing to one of the rotation dials components, and would DISABLE a vanilla feature from the other rotation dial component... Ugh, why does Scratch have 2 separate components for the same thing in the first place.

**Comment by BroJac5246** [comment]
> > It seems that the consensus is that modifier keys should trigger or disable this behavior (for example, no key pressed could result in 1-degree increments, Shift could result in 15-degree increments, and Ctrl could be for 45-degree increments). Of course, this behavior would be customizable.
> 
> Yeah, but one of the two rotation dial UIs in Scratch already snap every a certain amount of degrees... Would these settings apply to both?
> I mean, it's just weird that disabling the "snap" setting would do nothing to one of the rotation dials components, and would DISABLE a vanilla feature from the other rotation dial component... Ugh, why does Scratch have 2 separate components for the same thing in the first place.

Ideally it should apply to both because they do the same thing.

As for disabling a vanilla feature, we're really just making it more customable and consistent at the same time. Two things for the price of one!
